File: ios/VergiPro/Core/Data/ResourceState.swift
import Foundation

enum ResourceState<Value: Sendable>: Sendable {
    case idle
    case loading(previous: Value?)
    case content(Value, isStale: Bool)
    case empty
    case failure(APIError, previous: Value?)
}

struct Page<Value: Sendable>: Sendable {
    let items: [Value]
    let nextCursor: String?
}

File: android/app/src/main/java/com/vergipro/mobile/core/data/ResourceState.kt
package com.vergipro.mobile.core.data

import com.vergipro.mobile.core.network.ApiFailure

sealed interface ResourceState<out T> {
    data object Idle : ResourceState<Nothing>
    data class Loading<T>(val previous: T? = null) : ResourceState<T>
    data class Content<T>(val value: T, val isStale: Boolean = false) : ResourceState<T>
    data object Empty : ResourceState<Nothing>
    data class Failure<T>(val error: ApiFailure, val previous: T? = null) : ResourceState<T>
}

data class Page<T>(val items: List<T>, val nextCursor: String?)
